rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java

```diff
@@ -144,13 +147,110 @@ void testGetOldFederation(
         federationStorageProvider.setOldFederation(null);
         Federation oldFederation = federationStorageProvider.getOldFederation(federationConstants, activations);
         assertNull(oldFederation);
     }
 
+    private static Stream<Arguments> providePendingFederationAndFormatArguments() {
+        return Stream.of(
+            Arguments.of(P2SH_ERP_FEDERATION_FORMAT_VERSION, buildMockPendingFederation()),
+            Arguments.of(NON_STANDARD_ERP_FEDERATION_FORMAT_VERSION, buildMockPendingFederation()),
+            Arguments.of(STANDARD_MULTISIG_FEDERATION_FORMAT_VERSION, buildMockPendingFederation()),
+            Arguments.of(INVALID_FEDERATION_FORMAT, buildMockPendingFederation()),
+            Arguments.of(EMPTY_FEDERATION_FORMAT, buildMockPendingFederation())
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("providePendingFederationAndFormatArguments")
+    void testGetPendingFederation(
+        int federationFormat,
+        PendingFederation expectedFederation
+    ) {
+
+        // Arrange
+
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+
+        when(activations.isActive(ConsensusRule.RSKIP123)).thenReturn(false);
+
+        if(federationFormat != INVALID_FEDERATION_FORMAT && federationFormat != EMPTY_FEDERATION_FORMAT) {
+            when(activations.isActive(ConsensusRule.RSKIP123)).thenReturn(true);
+        }
+
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        byte[] federationFormatSerialized = getFederationFormatSerialized(federationFormat);
+        storageAccessor.saveToRepository(PENDING_FEDERATION_FORMAT_VERSION.getKey(), federationFormatSerialized);
+
+        byte[] serializedFederation = expectedFederation.serialize(activations);
+        System.out.println("serializedFederation: " + ByteUtil.toHexString(serializedFederation));
+        storageAccessor.saveToRepository(PENDING_FEDERATION_KEY.getKey(), serializedFederation);
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+        // Act
+
+        PendingFederation obtainedFederation = federationStorageProvider.getPendingFederation();
+
+        // Directly saving a null federation in storage to then assert that the method returns the cached federation
+        storageAccessor.saveToRepository(PENDING_FEDERATION_KEY.getKey(), null);
+
+        // Assert
+
+        // Call the method again and assert the same cached federation is returned
+        assertEquals(obtainedFederation, federationStorageProvider.getPendingFederation());
+
+        if(federationFormat == INVALID_FEDERATION_FORMAT || federationFormat == EMPTY_FEDERATION_FORMAT) {
+            assertArrayEquals(expectedFederation.serialize(activations), obtainedFederation.serialize(activations));
+        } else {
+            assertEquals(expectedFederation, obtainedFederation);
+        }
```

Comment:
This is confusing, should it be 2 separate test methods?

Reply:
I agree

Reply:
agree too

Reply:
Updated.